targets/csharp/make.js

```diff
@@ -223,28 +253,33 @@ function getPropertyCsType(property, datatype, needOptional) {
         throw "Unknown property type: " + property.actualtype + " for " + property.name + " in " + datatype.name;
 }
 
-function getAuthParams(apiCall) {
+function getAuthParams(apiCall, isApiInstance = false) {
     if (apiCall.url === "/Authentication/GetEntityToken")
         return "authKey, authValue";
     if (apiCall.auth === "EntityToken")
         return "\"X-EntityToken\", PlayFabSettings.EntityToken";
-    if (apiCall.auth === "SecretKey")
+    if (apiCall.auth === "SecretKey" && isApiInstance === false)
         return "\"X-SecretKey\", PlayFabSettings.DeveloperSecretKey";
+    if (apiCall.auth === "SecretKey" && isApiInstance === true)
+        return "\"X-SecretKey\", developerSecretKey";
+
     else if (apiCall.auth === "SessionTicket")
         return "\"X-Authorization\", PlayFabSettings.ClientSessionTicket";
     return "null, null";
 }
 
-function getRequestActions(tabbing, apiCall) {
+function getRequestActions(tabbing, apiCall, isApiInstance = false) {
     if (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest")
         return tabbing + "request.TitleId = PlayFabSettings.TitleId ?? request.TitleId;\n"
             + tabbing + "if (request.TitleId == null) throw new PlayFabException(PlayFabExceptionCode.TitleNotSet, \"Must be have PlayFabSettings.TitleId set to call this method\");\n";
     if (apiCall.auth === "EntityToken")
         return tabbing + "if (PlayFabSettings.EntityToken == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n";
     if (apiCall.auth === "SessionTicket")
         return tabbing + "if (PlayFabSettings.ClientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n";
-    if (apiCall.auth === "SecretKey")
+    if (apiCall.auth === "SecretKey" && isApiInstance === false)
         return tabbing + "if (PlayFabSettings.DeveloperSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"Must have PlayFabSettings.DeveloperSecretKey set to call this method\");\n";
+    if (apiCall.auth === "SecretKey" && isApiInstance === true)
+        return tabbing + "var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n            if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n";
```

Comment:
I get a build failure on release config because this is not protected with `#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API`

Reply:
Good catch. Please submit a fix.

Reply:
![image](https://user-images.githubusercontent.com/4026209/52895123-e2cecc80-31c6-11e9-9b54-c0305615556b.png)
Here is a bunch of code which is taken from multi-user-instance-support branch. We do this macro sets.  When we merge it will be ok. Should I make an other submit for fix

Reply:
this part of code should only work with Instanceable apis and  developer secret keys should be there. But To prevent an error which can be occur later I will add this flags.

 @pgilmorepf is there a problem if I add them in the multi-user-instance-support branch which is reviewing by  you 

@M-Pixel  on the other hand both release and debug configuration builds without any error on my side so which branch and solution(client,server,admin) you are working on ?  Can you send me screenshot of error to make understand where error occured